### ISSUE TITLE
info about ack'ed failovers not consistent between gui and backend db

### ISSUE THREAD

**Issue by igroene** [openark/orchestrator#414]
this is a follow up to #410 to try to debug why the gui and backend db don't appear to match.
I follow these steps:

1. kill master and let failover happen
2. leave the recovery unacked
3. kill the new master
4. ack the old failover via GUI
5. verify the old failover is displayed as acked via GUI on audit -> recovery page
7. the old failover is still shown not acked on backend db

```
sqlite> select * from topology_recovery where acknowledged=0;
               recovery_id = 14
                  hostname = mysql1
                      port = 3306
          in_active_period = 1
       start_active_period = 2018-02-15 16:16:31
end_active_period_unixtime = 0
              end_recovery = 2018-02-15 16:16:33
  processing_node_hostname = mysql1
    processcing_node_token = cea2f791b960ee95e45d8b654fd73ca8a2a4d8f468bcb02c59913057240974bd
        successor_hostname = mysql2
            successor_port = 3306
                  analysis = DeadMaster
              cluster_name = mysql1:3306
             cluster_alias = testcluster
     count_affected_slaves = 1
               slave_hosts = mysql2:3306
             is_successful = 1
              acknowledged = 0
           acknowledged_by =
       acknowledge_comment =
   participating_instances =
               lost_slaves =
                all_errors =
           acknowledged_at =
         last_detection_id = 34
           successor_alias =
                       uid = 1518711391799136695:855674e9c3c0073035e436d5c2f997c47e3d484b4b76455edc99f097021904d0
```
```
sqlite> select * from topology_recovery where in_active_period=1;
               recovery_id = 14
                  hostname = mysql1
                      port = 3306
          in_active_period = 1
       start_active_period = 2018-02-15 16:16:31
end_active_period_unixtime = 0
              end_recovery = 2018-02-15 16:16:33
  processing_node_hostname = mysql1
    processcing_node_token = cea2f791b960ee95e45d8b654fd73ca8a2a4d8f468bcb02c59913057240974bd
        successor_hostname = mysql2
            successor_port = 3306
                  analysis = DeadMaster
              cluster_name = mysql1:3306
             cluster_alias = testcluster
     count_affected_slaves = 1
               slave_hosts = mysql2:3306
             is_successful = 1
              acknowledged = 0
           acknowledged_by =
       acknowledge_comment =
   participating_instances =
               lost_slaves =
                all_errors =
           acknowledged_at =
         last_detection_id = 34
           successor_alias =
                       uid = 1518711391799136695:855674e9c3c0073035e436d5c2f997c47e3d484b4b76455edc99f097021904d0
```
```
sqlite> select hostname, port, cluster_name, alias from database_instance left join cluster_alias using (cluster_name);
    hostname = mysql2
        port = 3306
cluster_name = mysql2:3306
       alias = mysql2:3306

    hostname = mysql1
        port = 3306
cluster_name = mysql2:3306
       alias = mysql2:3306

    hostname = mysql3
        port = 3306
cluster_name = mysql2:3306
       alias = mysql2:3306
```
This is shown via GUI:
![image](https://user-images.githubusercontent.com/9805809/36267985-19ef5c22-1254-11e8-9eaf-269b18b3ea9a.png)

**Comment by shlomi-noach** [comment]
Can you please clarify which master you killed and when?

**Comment by igroene** [comment]
I first killed mysql1 around 16:16 and that caused mysql2 to become the master. After that I killed mysql2 and at that point the previous unacked failover no longer shows in the GUI.

**Comment by shlomi-noach** [comment]
I don't see the audit for the `mysql1` death in your screenshot. Did you `Audit -> Recovery` to get there or are you filtering by `mysql2:3306` cluster?

**Comment by shlomi-noach** [comment]
My suspicion at this time is that the link on the "recovery is blocked due to a previous failover..." alert leads you to the filtered list, and you really want the full list, or at least to see he unacknowledged failure. Please confirm if this is the case: that you got to the recovery-audit screen by clicking said link, and that if you get there via `Audit -> Recovery` you do see the `mysql1:3306` failure.

**Comment by shlomi-noach** [comment]
I'm able to reproduce, please hold on.

**Comment by shlomi-noach** [comment]
Released in `3.0.7`

**Comment by igroene** [comment]
Thank you!!
